projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -124,12 +134,16 @@ export class ExplorerComponent {
     this.filters = [...newFilters];
   }
 
-  public onContextUpdated(label: ContextLabel): void {
-    this.context = this.contextLabelToDashboardContext(label);
-    // Set query param async to allow any initating route change to complete
+  private getContextItem(label: string): ToggleItem<ContextLabel> | undefined {
+    return this.contextItems.find(item => item.label === label);
+  }
+
+  public onContextUpdated(item: ToggleItem<ContextLabel> = this.contextItems[0]): void {
```

Comment:
This should just take a context. It would remove the `value!`s we see below, and the label isn't relevant here.